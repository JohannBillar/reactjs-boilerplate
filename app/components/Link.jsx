import React, { PropTypes, Component } from 'react';

const STATUS = {
  NORMAL: 'normal',
  HOVERED: 'hovered',
};

const propTypes = {
  page: PropTypes.string,
  children: PropTypes.node,
};

class Link extends Component {
  constructor() {
    super();

    this.handleMouseEnter = this.handleMouseEnter.bind(this);
    this.handleMouseLeave = this.handleMouseLeave.bind(this);

    this.state = {
      class: STATUS.NORMAL,
    };
  }

  handleMouseEnter() {
    this.setState({ class: STATUS.HOVERED });
  }

  handleMouseLeave() {
    this.setState({ class: STATUS.NORMAL });
  }

  render() {
    return (
      <a
        className={this.state.class}
        href={this.props.page || '#'}
        onMouseEnter={this.handleMouseEnter}
        onMouseLeave={this.handleMouseLeave}
      >
        {this.props.children}
      </a>
    );
  }
}

Link.propTypes = propTypes;

export default Link;
