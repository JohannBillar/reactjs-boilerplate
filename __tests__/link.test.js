'use-strict';

import React from 'react';
import Link from '../app/components/Link';
import renderer from 'react-test-renderer';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <Link page="http://estudio.io/">Estudio</Link>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onMouseEnter();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onMouseLeave();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
