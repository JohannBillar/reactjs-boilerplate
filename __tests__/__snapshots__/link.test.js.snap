exports[`test Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="http://estudio.io/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Estudio
</a>
`;

exports[`test Link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="http://estudio.io/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Estudio
</a>
`;

exports[`test Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="http://estudio.io/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  Estudio
</a>
`;
